apps/devhub/views.py

```diff
@@ -1210,16 +1210,17 @@ def version_add(request, addon_id, addon):
                       args=[addon.slug, str(version.id)])
 
         # Not listed? Do we need to sign?
-        if (waffle.flag_is_active(request, 'automatic-validation') and
-                not addon.is_listed):
-            # We need to get to the file to check its validation.
-            # We're submitting a new version here, so it has only one file.
+        if waffle.flag_is_active(request, 'automatic-validation'):
             file_ = version.files.get()
-            # Only if it's not sideload (not fully reviewed).
-            if (addon.status not in [amo.STATUS_NOMINATED, amo.STATUS_PUBLIC]
-                    and not file_.validation.errors
-                    and not file_.validation.warnings):
-                # Passed validation: sign automatically without manual review.
+            validation_passed = bool(not file_.validation.errors and
+                                     not file_.validation.warnings)
+            if addon.is_listed and is_beta and validation_passed:
```

Comment:
I think we need a special case here in case the request was crafted to submit the file as beta and the validation didn't pass (raising a `PermissionDenied`?):
```
if addon.is_listed and is_beta:
    if validation_passed:
        sign_file(file_)
    else:
        raise PermissionDenied
```
(and the associated test)